src/framework/global/io/file.cpp

```diff
@@ -86,23 +86,27 @@ bool File::remove()
 bool File::doOpen(OpenMode m)
 {
     if (m == IODevice::WriteOnly) {
-        //! NOTE Not need fetch data, file will be truncated
-        return true;
+        Ret ret = fileSystem()->isWritable(m_filePath);
+        if (!ret) {
+            setError(ret.code(), ret.text());
+        }
+
+        return ret;
     }
 
     if (!exists()) {
         if (m == OpenMode::ReadOnly) {
-            m_error = Error::ReadError;
+            setError(int(Err::FSReadError), "Opening non-existent file called on a read-only mode");
```

Comment:
Shouldn't this be translatable?

Reply:
Also the string shpuld IMHO rather be "Opening non-existent file in a read-only mode"